tools/mktutorial/main.go

```diff
@@ -82,15 +82,50 @@ func main() {
 	}
 }
 
+// getPulumiYamlPath returns the path to the folder in a given tutorial,
+// that contains the Pulumi.yaml file. It does not recursively search for it, though.
+func getPulumiYamlPath(root, tutorialName string) string {
```

Comment:
I wonder if instead we should determine whether we should show the button based on if it's present in the README.md.

Basically search the README.md for:
 - `[![Deploy](https://get.pulumi.com/new/button.svg)](https://app.pulumi.com/new)` - use the root tutorial dir for the `template` query param
 - `[![Deploy](https://get.pulumi.com/new/button.svg)](https://app.pulumi.com/new?template=<url>)` - use the specified `<url>` for the `template` query param

If we can't find the button in the README.md, don't show it on the docs site.

Reply:
I am fine changing it to use this convention instead.